File: deaf_grandma.js
let goodbyeCount = 0;
// let convo = true;

// function deafGrandma(input) {
//   console.log(goodbyeCount);
function deafGrandma() {
  let input = window.prompt("Answer Grandma!");
  let lower = input.toLowerCase();

  if (input.length === 0) {
    alert("WHAT?!");
    deafGrandma();
  } else if (input.includes(lower)) {
    alert("SPEAK UP, KID!");
    deafGrandma();
  } else if (input === input.toUpperCase() && input != "GOODBYE!") {
    alert("NO, NOT SINCE 1946!");
    deafGrandma();
  } else if (input === "GOODBYE!" && goodbyeCount === 0) {
    goodbyeCount++;
    alert("LEAVING SO SOON?");
    deafGrandma();
  } else if (input === "GOODBYE!" && goodbyeCount > 0) {
    alert("LATER SKATER!");
    // convo = false;
  }
}
// }
deafGrandma();

File: deaf_grandma.py
import re

goodbye_count = 0;

def deaf_grandma():
  answer = input("Answer Grandma!");
#   lower = answer.lower();
  global goodbye_count
  nothing = ''

  if answer == nothing:
    print("WHAT?!");
    deaf_grandma();
  elif re.search("[a-z]", answer):
    print("SPEAK UP, KID!");
    deaf_grandma();
  elif re.search("[A-Z]", answer) and answer != "GOODBYE!":
    print("NO, NOT SINCE 1946!");
    deaf_grandma();
  elif answer == "GOODBYE!" and goodbye_count == 0:
    goodbye_count = 1;
    print("LEAVING SO SOON?")
    deaf_grandma()
  elif answer == "GOODBYE!" and goodbye_count > 0:
    print("LATER SKATER!");
  


deaf_grandma()
